src/core/libmaven/databases.cpp

```diff
@@ -213,8 +213,19 @@ bool Databases::addCompound(Compound* c) {
         for(unsigned int i = 0; i < compoundsDB.size(); i++) {
             Compound* currentCompound = compoundsDB[i];
             if ( currentCompound->db() == c->db() && currentCompound->id() == c->id()) {
-                currentCompound = c;
```

Comment:
Why was the assignment not sufficient? The fix makes the same changes, but assigns individual attributes instead. The original one-liner looked more elegant. Maybe it is because the assignment does not take care of all the members of a compound? I mentioned this in another comment earlier.

Reply:
umm.. Actually, I tried doing this. Assigning all the values but the thing is  currentCompound is getting its value from compoundDB[i] and only specific attributes from 'c' are being set. Thus, using "currentCompound = c" is not giving the desired results.

Reply:
What are the desired results here? Because if we are assigning attributes of a compound to another one, then there should be a reason to do this only partially.

Reply:
Due to this assignment, original problem of isomers shows back that 'glucose-6-phosphate' doesn't come in the result set and instead 'fructose-6-phosphate' is shown